Build fails due to problems accessing the docker image for postgres

For a few days, all builds keep failing, e.g. https://github.com/ursjoss/scipamato/pull/259/checks?check_run_id=1852808211

<details><summary>Stacktrace</summary>
<pre>
Caused by: com.spotify.docker.client.exceptions.ImageNotFoundException: Image not found: postgres:10
	at com.spotify.docker.client.DefaultDockerClient.createContainer(DefaultDockerClient.java:747)
	at com.spotify.docker.client.DefaultDockerClient.createContainer(DefaultDockerClient.java:724)
	at ch.ayedo.jooqmodelator.core.Modelator.generate(Modelator.kt:37)
	at ch.ayedo.jooqmodelator.gradle.JooqModelatorTask.generateMetamodel(JooqModelatorTask.kt:78)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:104)
	... 119 more
Caused by: com.spotify.docker.client.exceptions.DockerRequestException: Request error: POST unix://localhost:80/containers/create: 404, body: {"message":"No such image: postgres:10"}

	at com.spotify.docker.client.DefaultDockerClient.propagate(DefaultDockerClient.java:2840)
	at com.spotify.docker.client.DefaultDockerClient.request(DefaultDockerClient.java:2723)
	at com.spotify.docker.client.DefaultDockerClient.createContainer(DefaultDockerClient.java:742)
	... 126 more
Caused by: javax.ws.rs.NotFoundException: HTTP 404 Not Found
	at org.glassfish.jersey.client.JerseyInvocation.convertToException(JerseyInvocation.java:1008)
	at org.glassfish.jersey.client.JerseyInvocation.translate(JerseyInvocation.java:816)
	at org.glassfish.jersey.client.JerseyInvocation.access$700(JerseyInvocation.java:92)
	at org.glassfish.jersey.client.JerseyInvocation$5.completed(JerseyInvocation.java:773)
	at org.glassfish.jersey.client.ClientRuntime.processResponse(ClientRuntime.java:198)
	at org.glassfish.jersey.client.ClientRuntime.access$300(ClientRuntime.java:79)
	at org.glassfish.jersey.client.ClientRuntime$2.run(ClientRuntime.java:180)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:271)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:267)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:315)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:297)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:267)
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:340)
	at org.glassfish.jersey.client.ClientRuntime$3.run(ClientRuntime.java:210)
</pre>
</details>

--- a/.github/workflows/check.yml
+++ b/.github/workflows/check.yml
@@ -36,5 +36,11 @@ jobs:
         restore-keys: |
           gradle-cache-${{ hashFiles('gradle/wrapper/gradle-wrapper.properties') }}
 
+    - name: Login to Docker Hub
+      uses: docker/login-action@v1
+      with:
+        username: ${{ secrets.DOCKERHUB_USERNAME }}
+        password: ${{ secrets.DOCKERHUB_TOKEN }}
+
     - name: Check with Gradle
       run: ./gradlew check --stacktrace

--- a/.github/workflows/check.yml
+++ b/.github/workflows/check.yml
@@ -43,7 +43,7 @@ jobs:
         password: ${{ secrets.DOCKERHUB_TOKEN }}
 
     - name: Download postgres
-      uses: docker://postgres:10
+      run: docker pull postgres:10
 
     - name: Check with Gradle
       run: ./gradlew check --stacktrace

--- a/.github/workflows/check.yml
+++ b/.github/workflows/check.yml
@@ -42,5 +42,8 @@ jobs:
         username: ${{ secrets.DOCKERHUB_USERNAME }}
         password: ${{ secrets.DOCKERHUB_TOKEN }}
 
+    - name: Download postgres
+      uses: docker://postgres:10
+
     - name: Check with Gradle
       run: ./gradlew check --stacktrace

--- a/.github/workflows/sonar.yml
+++ b/.github/workflows/sonar.yml
@@ -33,6 +33,9 @@ jobs:
         username: ${{ secrets.DOCKERHUB_USERNAME }}
         password: ${{ secrets.DOCKERHUB_TOKEN }}
 
+    - name: Download postgres
+      uses: docker://postgres:10
+
     - name: Run sonar task
       run: ./gradlew sonarqube --no-daemon
       env:

--- a/.github/workflows/sonar.yml
+++ b/.github/workflows/sonar.yml
@@ -34,7 +34,7 @@ jobs:
         password: ${{ secrets.DOCKERHUB_TOKEN }}
 
     - name: Download postgres
-      uses: docker://postgres:10
+      run: docker pull postgres:10
 
     - name: Run sonar task
       run: ./gradlew sonarqube --no-daemon

--- a/.github/workflows/sonar.yml
+++ b/.github/workflows/sonar.yml
@@ -27,6 +27,12 @@ jobs:
         restore-keys: |
           gradle-cache-${{ hashFiles('gradle/wrapper/gradle-wrapper.properties') }}
 
+    - name: Login to Docker Hub
+      uses: docker/login-action@v1
+      with:
+        username: ${{ secrets.DOCKERHUB_USERNAME }}
+        password: ${{ secrets.DOCKERHUB_TOKEN }}
+
     - name: Run sonar task
       run: ./gradlew sonarqube --no-daemon
       env:

